modules/client/src/controllers/ConditionalTransferController.ts

```diff
@@ -207,4 +210,24 @@ export class ConditionalTransferController extends AbstractController {
   private conditionalExecutors: ConditionalExecutors = {
     LINKED_TRANSFER: this.handleLinkedTransfers,
   };
+
+  private async waitForAppInstall(): Promise<void> {
+    return new Promise(
+      async (res: any, rej: any): Promise<any> => {
+        const getAppIds = async (): Promise<string[]> => {
+          return (await this.connext.getAppInstances()).map((a: AppInstanceInfo) => a.identityHash);
+        };
+        let retries = 0;
+        while (!(await getAppIds()).includes(this.appId) && retries <= MAX_RETRIES) {
```

Comment:
These are the kind of pollers that we'll be able to remove once finished events are published right?

Reply:
Yep, we will do this before merging (or maybe in a separate PR if we get these tests passing first).